src/sourcemap/sourcemap.zig

```diff
@@ -449,6 +449,16 @@ pub const LineColumnOffset = struct {
             var iter = strings.CodepointIterator.initOffset(input, i);
             var cursor = strings.CodepointIterator.Cursor{ .i = @as(u32, @truncate(iter.i)) };
             _ = iter.next(&cursor);
+
+            // Given a null byte, cursor.width becomes 0
+            // This can lead to integer overflow, crashes, or hangs.
+            // https://github.com/oven-sh/bun/issues/10624
+            if (cursor.width == 0) {
+                columns += 1;
+                offset = i + 1;
+                continue;
+            }
```

Comment:
I think we should use the return value of `iter.next` (which returns `false` on a byte sequence length of 0) to detect the invalid case, e.g.
```suggestion
            // Given a null byte, cursor.width becomes 0
            // This can lead to integer overflow, crashes, or hangs.
            // https://github.com/oven-sh/bun/issues/10624
            if (!iter.next(&cursor)) {
                columns += 1;
                offset = i + 1;
                continue;
             }            
```
This would also make the code less likely to break if we change the default `width` value in `CodepointIterator.Cursor`

Reply:
.next() returns false when it reaches the end

It shouldn't advance columns by 1 when the end is reached

Reply:
makes sense 👍